lib/pliny/librato/metrics/backend.rb

```diff
@@ -1,49 +1,48 @@
 require 'librato/metrics'
 require 'pliny/error_reporters'
+require 'librato/collector'
 
 module Pliny
   module Librato
     module Metrics
       # Implements the Pliny::Metrics.backends API. Puts any metrics sent
       # from Pliny::Metrics onto a queue that gets submitted in batches.
       class Backend
-        POISON_PILL = :'❨╯°□°❩╯︵┻━┻'
-
         def initialize(source: nil, interval: 60, count: 500)
           @interval      = interval
           @mutex         = Mutex.new
-          @metrics_queue = Queue.new
+          @counter_cache = ::Librato::Collector::CounterCache.new(default_tags: nil)
```

Comment:
I'm guessing this is an in-memory cache? What happens if the thread dies, do we lose all cached metrics?